easydock/protonation.py

```diff
@@ -27,3 +32,32 @@ def read_protonate_chemaxon(fname):
             smi = mol.GetPropsAsDict().get('MAJORMS', None)
             if smi is not None:
                 yield smi, mol_name
+
+def chunk_into_n_file(fname, number_of_files):
+    with open(fname) as infp:
+        fname_list = [fname[:-4]+'_input'+str(i)+'.smi' for i in range(number_of_files)]
+        files = [open(fname[:-4]+'_input'+str(i)+'.smi','w') for i in range(number_of_files)]
+        for i, line in enumerate(infp):
+            files[i % number_of_files].write(line)
+        for f in files:
+            f.close()
+        return fname_list
+
+def dummy_protonate_dimorphite(input_fname):
+    dimorphite_run(smiles_file=input_fname, output_file=input_fname[:-4]+'_output.smi', max_variants=1, silent=True, min_ph=7.4, max_ph=7.4)
+
+def protonate_dimorphite(input_fname, output_fname, ncpu):
+    
+    pool = Pool(ncpu)
+    fname_list = chunk_into_n_file(input_fname, ncpu)
+    
+    with open(output_fname, 'wt') as output_file:
+        pool.map(dummy_protonate_dimorphite, fname_list)
+
+        for fname in fname_list:
+            output_file.writelines(open(fname[:-4]+'_output.smi','r').readlines())
```

Comment:
I suggest to open files with `with` statement, otherwise those file descriptors will not be closed. This is always a general recommendation.